starter-code/public/scripts/models/repo.js

```diff
@@ -7,16 +7,37 @@ var app = app || {};
   repos.all = [];
 
   repos.requestRepos = function(callback) {
-    // TODO: How would you like to fetch your repos? Don't forget to call the callback.
+    // DONE: How would you like to fetch your repos? Don't forget to call the callback.
     //       Remember that the callback function we'll want to call relies on repos.all
     //       being an array with a bunch of repo objects in it, so you'll need to
     //       populate it with the response from Github before you call the callback.
-
-  };
+    // estimated 20 min || actual 30 min
+    $.ajax({
+      url: `https://api.github.com/user/repos`,
+      method: 'GET',
+      headers: {
+        Authorization: `token ${githubToken}`
+      }
+    })
+    .then(
+      function(data){
+        repos.all = data.map(repo => ({
+          name: repo.name,
+          html_url: repo.html_url,
+          description: repo.description,
+          languages: repo.languages_url,
+          created_at: repo.created_at,
+          updated_at: repo.updated_at,
+          watchers_count: repo.watchers_count
+        }
+      ));
+      }
+    )
+    callback();
```

Comment:
Your callback is in the wrong place. It is outside your .then and it should either be inside the .then or it should be in its own .then